### PR TITLE
Gaussian accelerated billiard walk for volume cooling ellipsoids algorithm.

### PR DESCRIPTION
**Pull request by vaithak**
Most of the changes in the PR have been pulled apart from #171 for easy reviewal and better readability.

### CODE REVIEW

**Comment by vissarion** [comment]
Thanks, are there any new changes compared to #171 or this PR is a subset of #171?

**Comment by vaithak** [comment]
these changes are a subset of #171.

File: external/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_deps/
```
Comment (vissarion) [inline external/.gitignore]: is this still needed?

Comment (vaithak) [reply external/.gitignore]: removed 👍🏽

**Review by TolisChal — APPROVED** [review]

**Review by vissarion — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**